independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/DirectoryPathTree.java

```diff
@@ -73,20 +74,33 @@ public void accept(String relativePath, Consumer<PathVisit> consumer) {
             consumer.accept(null);
             return;
         }
-        final Path path = dir.resolve(manifestEnabled ? toMultiReleaseRelativePath(relativePath) : relativePath);
+        final Path path = dir.resolve(interpretPath(relativePath, manifestEnabled));
         if (!Files.exists(path)) {
             consumer.accept(null);
             return;
         }
         PathTreeVisit.consume(dir, dir, path, pathFilter, consumer);
     }
 
+    private String interpretPath(String relativePath, boolean manifestEnabled) {
+        FileSystem fileSystem = dir.getFileSystem();
+        Path path = fileSystem.getPath(relativePath);
+        if (path.isAbsolute()) {
+            // We got passed an absolute path; interpret it as relative to `dir`.
+            // Without this, we would be allowing callers to access the whole filesystem tree,
+            // because absolute paths would not necessarily be pointing to files under `dir`.
+            path = path.relativize(path.getRoot());
+            relativePath = path.toString();
+        }
```

Comment:
Thanks @yrodiere There are a few things I wanted to improve here though.
The "sanitization" should happen earlier to make sure the path is visible through the filters.
The path could be relative but still under the root dir.
Given that this method will be called a lot during bootstrap and classloading, I would like to avoid creating `Path` instances where possible, taking into account that absolute and non-normalized paths it will be very rare.
I tried to do that in https://github.com/quarkusio/quarkus/pull/23692 It's still not perfect though.